WeatherDataFetcher.py

```diff
@@ -1,101 +1,166 @@
+            file=sys.stderr,
         )
+        return fetch_location_coords(input("Where do you live?\n-->"))
+    location_coords = (location.latitude, location.longitude)
+    return location_coords
+
 
-        file_path_json = os.getcwd() + "/data.json"
-        if reload or not os.path.exists(file_path_json):
-            with urllib.request.urlopen(request) as url:
-                data = json.loads(url.read().decode())
-                with open(file_path_json, "w", encoding="utf-8") as f:
-                    json.dump(data, f, ensure_ascii=False, indent=4)
-        else:
-            with open(file_path_json, "r", encoding="utf-8") as f:
-                data = json.loads(f.read())
-
-        for _, values in enumerate(data["list"]):
-            timestamp_utc = values["dt"]
-            timestamp = datetime.datetime.utcfromtimestamp(timestamp_utc)
-            temp_min = values["main"]["temp_min"]
-            temp_max = values["main"]["temp_max"]
-            print(
-                f"On:{timestamp}, the min Temp is: {temp_min} and the max Temp is {temp_max}"
-            )
-
-
-class Location_Coords_Fetcher:
+def get_api_key() -> str:
     """
-    Class for getting location coordinates (long and lat) for a given city
-    Parameters
-    ---------
-    city : str
-      Name of the city for the coordinates
-    location_coords : Tuple
-      (lat, long) for the self.city
+    Checks, if API Key is set as an environment variable. If not, the user is
+    asked to input it in the console. Length checks enabled for the API key. If
+    key has non-valid length (!=32) the user is asked again to enter a valid key.
+    The key is saved into a local .env file.
+
+    Returns:
+    api_key : str
+      API Key for the Open Weather Map
+
     """
 
-    def __init__(self):
-        self.location_coords = self.fetch_location_coords()
+    dotenv_file = dotenv.find_dotenv()
+    dotenv.load_dotenv(dotenv_file)
 
-    def fetch_location_coords(self) -> Tuple:
-        # ask user for city
-        self.city = input("Where do you live?\n-->")
-        # city = "Berlin"
-        geolocator = Nominatim(user_agent="MyApp")
-        location = geolocator.geocode(self.city)
-        if location is None:
-            print(
-                f"Sorry, your input for the city ({self.city}) was not found, try again:"
-            )
-            return self.fetch_location_coords()
-        location_coords = (location.latitude, location.longitude)
-        return location_coords
+    if "TNT_EX1_OPENWEATHERMAP_API_KEY" not in os.environ:
+        print(
+            "No API Key found in your environment variables. \nPlease look at https://openweathermap.org/api for getting an API key and enter it in the following line:",
+            file=sys.stderr,
+        )
+        os.environ["TNT_EX1_OPENWEATHERMAP_API_KEY"] = input(
+            "Please enter your API Key now: \n-->"
+        ).strip()
+    api_key = os.environ["TNT_EX1_OPENWEATHERMAP_API_KEY"]
```

Comment:
That is a bit odd logic. You could just set it above and not set it in-between as an environment var.